File: backend/src/Core/Bookify.Domain/Entities/Professional.cs
// path: backend/src/Core/Bookify.Domain/Entities/Professional.cs

namespace Bookify.Domain.Entities;

public class Professional
{
    public Guid Id { get; init; }
    public string Name { get; set; } = string.Empty;
    public string Specialty { get; set; } = string.Empty;
    public string Bio { get; set; } = string.Empty;
    public string AvatarUrl { get; set; } = string.Empty;
}

File: backend/src/main/java/com/psychescheduler/backend/domain/Professional.java
package com.psychescheduler.backend.domain;


import jakarta.persistence.*;
import lombok.*;

import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity




@Table(name = "professionals")
public class Professional {

    @Id
    private UUID id;

    @Column(nullable = false, length = 200)
    private String name;

    @Column(nullable = false, length = 200)
    private String specialty;


    @Lob
    @Column(columnDefinition = "TEXT")
    private String bio;

    @Column(name = "avatar_url")
    private String avatarUrl;

    @OneToOne
    @JoinColumn(name= "user_id", referencedColumnName = "id")
    private User user;
}
